eventuate-tram-reactive-consumer-jdbc/src/main/java/io/eventuate/tram/reactive/consumer/jdbc/ReactiveTransactionalNoopDuplicateMessageDetector.java

```diff
@@ -21,7 +19,7 @@ public ReactiveTransactionalNoopDuplicateMessageDetector(TransactionalOperator t
   }
 
   @Override
-  public Supplier<Mono<Void>> doWithMessage(SubscriberIdAndMessage subscriberIdAndMessage, Supplier<Mono<Void>> processingFlow) {
-    return () -> processingFlow.get().as(transactionalOperator::transactional);
+  public Mono<Void> doWithMessage(SubscriberIdAndMessage subscriberIdAndMessage, Mono<Void> processingFlow) {
+    return Mono.defer(() -> processingFlow.as(transactionalOperator::transactional));
```

Comment:
should it be 

return Mono.defer(() -> processingFlow).as(transactionalOperator::transactional);

Reply:
Thank you, I will check

Reply:
I suspect that this change is not necessary. Also transaction processing should be deferred until it's needed.